packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -449,210 +528,170 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
-              const expectedCost = await getExpectedCostOfLastLargeTx(contract.deploymentTransaction()!.data);
-
-              // awaiting for HBAR limiter to finish updating expenses in the background
-              await Utils.wait(6000);
-
-              const spendingPlan = await hbarSpendingPlanRepository.findByIdWithDetails(
-                hbarSpendingPlan.id,
-                requestDetails,
-              );
+              await contract.waitForDeployment();
+              const expectedTxCost = await getExpectedCostOfLastLargeTx(contract.deploymentTransaction()!.data);
 
-              const amountSpent = spendingPlan.amountSpent;
+              const amountSpent = await pollForProperAmountSpent(hbarSpendingPlan, 1, expectedTxCost);
 
               const tolerance = 0.01;
-              expect(amountSpent).to.be.approximately(expectedCost, tolerance * expectedCost);
+              expect(amountSpent).to.be.approximately(expectedTxCost, tolerance * expectedTxCost);
             });
 
-            it(`Should eventually exhaust the hbar limit for ${subsriptionTier} user`, async () => {
-              let deploymentCounts: number = 0;
-              const exchangeRateResult = (await mirrorNode.get(`/network/exchangerate`, requestId)).current_rate;
-              const exchangeRateInCents = exchangeRateResult.cent_equivalent / exchangeRateResult.hbar_equivalent;
+            it(`Should eventually exhaust the hbar limit for ${subscriptionTier} user and still allow another ${subscriptionTier} user to make calls`, async () => {
+              let expectedTxCost = 0;
+              let deploymentCounts = 0;
               const remainingHbarsBefore = Number(await metrics.get(testConstants.METRICS.REMAINING_HBAR_LIMIT));
 
-              const factory = new ethers.ContractFactory(
-                largeContractJson.abi,
-                largeContractJson.bytecode,
-                accounts[0].wallet,
-              );
-              const deployedTransaction = await factory.getDeployTransaction();
-              const estimatedTxFee = estimateFileTransactionsFee(
-                deployedTransaction.data.length,
-                fileAppendChunkSize,
-                exchangeRateInCents,
-              );
-
               try {
                 for (deploymentCounts = 0; deploymentCounts < 50; deploymentCounts++) {
-                  await deployContract(largeContractJson, aliasAccount.wallet);
+                  const tx = await deployContract(largeContractJson, aliasAccount.wallet);
+                  await tx.waitForDeployment();
+
+                  expectedTxCost ||= await getExpectedCostOfLastLargeTx(largeContractJson.bytecode);
+                  await pollForProperAmountSpent(hbarSpendingPlan, deploymentCounts + 1, expectedTxCost);
                 }
                 expect.fail(`Expected an error but nothing was thrown`);
               } catch (e: any) {
+                logger.error(e.message);
```

Comment:
Q: why `console` vs `logger`?

Reply:
Ah yeah I used `logger` first for consistency as the whole suite uses it. The second and main reason is to maintain asynchronous order. With `console`, logs appear immediately upon execution, but with `logger`, logs wait for other `logger`-initiated entries. This creates a more organized order, making it easier to navigate through logs during debugging.

Reply:
Yeah I've observed this also.
I wonder if this means other logs should be converted for easier troubleshooting